sycl/source/detail/jit_compiler.cpp

```diff
@@ -153,7 +153,8 @@ struct PromotionInformation {
 
 using PromotionMap = std::unordered_map<SYCLMemObjI *, PromotionInformation>;
 
-template <typename Obj> Promotion getPromotionTarget(const Obj &obj) {
+template <typename Obj>
+Promotion getPromotionTarget([[maybe_unused]] const Obj &obj) {
```

Comment:
Can you not delete the parameter name instead? since it is actually unused.
same for the other change

Reply:
I agree with this, but didn't wanna be too picky

Reply:
We could, but we expect to need this parameter in another PR soonish, so this change seems less invasive and allows us to fix post-commit CI quicker.